src/sgame/sg_bot_parse.cpp

```diff
@@ -142,9 +142,38 @@ static AIValue_t haveUpgrade( gentity_t *self, const AIValue_t *params )
 	return AIBoxInt( !BG_UpgradeIsActive( upgrade, self->client->ps.stats ) && BG_InventoryContainsUpgrade( upgrade, self->client->ps.stats ) );
 }
 
-static AIValue_t percentAmmo( gentity_t *self, const AIValue_t* )
+// This function returns the ratio of 2 ammo values for a
+// specific weapon. Only used to reduce duplicated code.
+// TODO: make this a weapon_t or weaponAttributes_t method?
+static AIValue_t ratio( int current, int max, weapon_t wp )
 {
-	return AIBoxFloat( PercentAmmoRemaining( BG_PrimaryWeapon( self->client->ps.stats ), &self->client->ps ) );
+	if ( max == 0.f || BG_Weapon( wp )->infiniteAmmo )
+	{
+		return AIBoxFloat( 1.f );
+	}
+	return AIBoxFloat( static_cast<float>( current ) / static_cast<float>( max ) ); //TODO: only keep the useful one (I never remember which one it must be)
```

Comment:
What does this comment mean?

Reply:
it means that I never remember which damned side of a divide must be a float to not loose precision, and that I forgot to check and fix it before doing the PR :/

Reply:
It's a floating-point division if either side is floating point.